packages/vitest/src/node/cli/cli-api.ts

```diff
@@ -73,8 +73,9 @@ export async function startVitest(
     stdinCleanup = registerConsoleShortcuts(ctx, stdin, stdout)
   }
 
-  ctx.onServerRestart((reason) => {
+  ctx.onServerRestart(async (reason) => {
     ctx.report('onServerRestart', reason)
+    await ctx.close()
```

Comment:
I was looking at these specific lines earlier this week, and decided to move this logic into `core.ts` directly. We don't really need to attach this `onServerRestart` listener outside the ctx.

- https://github.com/AriPerkkio/vitest/commit/7e222e999314164e5a3d29c89897a1dc4bf6fdd9#diff-49ef635be88fe607c8682e81ab56b061ba9aafd5c94a5690a70b90a54604cd24L76-L79
- https://github.com/AriPerkkio/vitest/commit/7e222e999314164e5a3d29c89897a1dc4bf6fdd9#diff-c70ce04cc623628f74475cbde7bf56c29e5c3034b6fd5d26f737cb3944254463R160-R161

We could do this refactoring in this PR too.

Reply:
I don't really see any benefit in moving it to be honest 🤔 If we removed `ctx.onServerRestart` entirely - then maybe 🤷🏻

Reply:
As I was looking all the places where we call reporters, this line looked surprising. Why are we attaching these listeners outside `core.ts`? Why don't we call `ctx.report('onServerRestart', reason)` directly inside `core.ts`? If there's a real reason for these lines to be outside it, here in `cli-api`, then sure let's leave them.

Reply:
Sure, I don't have enough arguments against it